src/Service/NuRaftLogSnapshot.cpp

```diff
@@ -586,6 +602,13 @@ void KeeperSnapshotStore::loadLatestSnapshot(KeeperStore & store)
     thread_pool.wait();
     LOG_INFO(log, "Parsing snapshot objects costs {}ms", watch.elapsedMilliseconds());
 
+    if (load_objects_count && *load_objects_count != objects_path.size())
```

Comment:
Maybe it is better to move the checking in the front of loading data tree. We can load object1 first and then load data tree in parellel. And we can do it latter when squashing the first 3 objects.

Reply:
Checking load_objects_count here is an ugly implementation, but it avoids some code changes and refactoring. I will do it  when squashing the first 3 objects.